spec/support/govuk_i18n_cov.rb

```diff
@@ -0,0 +1,81 @@
+# frozen_string_literal: true
+
+class GOVUKI18nCov
+  module I18n
+    module Registry
+      def lookup(locale, key, scope = [], options = {})
+        GOVUKI18nCov.log(locale, key, scope)
+        super
+      end
+    end
+  end
+
+  class << self
+    def start
+      return if @started
+
+      @started = true
+      @used_keys = []
+      ::I18n::Backend::Simple.send :include, I18n::Registry
+    end
+
+    def log(locale, key, scope)
+      prefix = [locale] + Array(scope || [])
+      @used_keys << "#{prefix.compact.join('.')}.#{key}"
+    end
+
+    def report
+      return unless @started
+
+      used = normalise_keys(@used_keys).uniq
+      all = get_all_i18n_keys
+      unused = all.reject { |as| used.any? { |us| as.start_with?(us) } }
+
+      if unused.any?
+        write_report(unused)
+        print_message(all, unused)
+      end
+    end
+
+  private
+
+    def get_all_i18n_keys
+      ymls = Dir.glob("config/locales/**/*.yml")
```

Comment:
Should it use Rails to access these? We configure this here: https://github.com/alphagov/content-publisher/blob/57e6867e9033cef6c2c24f88e1ded8d7569de61e/config/application.rb#L32

Though after looking this up I see that it will include all the gems too so this might be better.

Reply:
You got it in one! I've hopefully made this a bit clearer, plus making it somewhat configurable to support testing.